lib/internal/crypto/webcrypto.js

```diff
@@ -0,0 +1,740 @@
+  validateBoolean(extractable, 'extractable');
+  validateArray(keyUsages, 'keyUsages');
+  switch (algorithm.name) {
+    case 'RSASSA-PKCS1-V1_5':
+      // Fall through
+    case 'RSA-PSS':
+      // Fall through
+    case 'RSA-OAEP':
+      return lazyRequire('internal/crypto/rsa')
+        .rsaImportKey(format, keyData, algorithm, extractable, keyUsages);
+    case 'ECDSA':
+      // Fall through
+    case 'ECDH':
+      return lazyRequire('internal/crypto/ec')
+        .ecImportKey(format, keyData, algorithm, extractable, keyUsages);
+    case 'HMAC':
+      return lazyRequire('internal/crypto/mac')
+        .hmacImportKey(format, keyData, algorithm, extractable, keyUsages);
+    case 'AES-CTR':
+      // Fall through
+    case 'AES-CBC':
+      // Fall through
+    case 'AES-GCM':
+      // Fall through
+    case 'AES-KW':
+      return lazyRequire('internal/crypto/aes')
+        .aesImportKey(algorithm, format, keyData, extractable, keyUsages);
+    case 'HKDF':
+      // Fall through
+    case 'NODE-SCRYPT':
+      // Fall through
+    case 'PBKDF2':
+      return importGenericSecretKey(
+        algorithm.name,
+        format,
+        keyData,
+        extractable,
+        keyUsages);
+    case 'NODE-DSA':
+      return lazyRequire('internal/crypto/dsa')
+        .dsaImportKey(format, keyData, algorithm, extractable, keyUsages);
+    case 'NODE-DH':
+      return lazyRequire('internal/crypto/diffiehellman')
+        .dhImportKey(format, keyData, algorithm, extractable, keyUsages);
+  }
+
+  throw lazyDOMException('Unrecognized name.', 'NotSupportedError');
+}
+
+// subtle.wrapKey() is essentially a subtle.exportKey() followed
+// by a subtle.encrypt().
+async function wrapKey(format, key, wrappingKey, algorithm) {
+  algorithm = normalizeAlgorithm(algorithm);
+  let keyData = await exportKey(format, key);
+
+  if (format === 'jwk') {
+    if (keyData == null || typeof keyData !== 'object')
+      throw lazyDOMException('Invalid exported JWK key', 'DataError');
+    const ec = new TextEncoder();
+    keyData = ec.encode(JSONStringify(keyData));
+  }
+
+  return cipherOrWrap(
+    kWebCryptoCipherEncrypt,
+    algorithm,
+    wrappingKey,
+    keyData,
+    'wrapKey');
+}
+
+// subtle.unwrapKey() is essentially a subtle.decrypt() followed
+// by a subtle.importKey().
+async function unwrapKey(
+  format,
+  wrappedKey,
+  unwrappingKey,
+  unwrapAlgo,
+  unwrappedKeyAlgo,
+  extractable,
+  keyUsages) {
+  wrappedKey = getArrayBufferOrView(wrappedKey, 'wrappedKey');
+
+  let keyData = await cipherOrWrap(
+    kWebCryptoCipherDecrypt,
+    normalizeAlgorithm(unwrapAlgo),
+    unwrappingKey,
+    wrappedKey,
+    'unwrapKey');
+
+  if (format === 'jwk') {
+    // The fatal: true option is only supported in builds that have ICU.
+    const options = process.versions.icu !== undefined ?
+      { fatal: true } : undefined;
+    const dec = new TextDecoder('utf-8', options);
+    try {
+      keyData = JSONParse(dec.decode(keyData));
```

Comment:
```suggestion
      keyData = JSONParse(TextDecoderPrototypeDecode(dec, keyData));
```
(if this isn't a thing, it should be)

Reply:
It's not a thing given that TextDecoder is not a primordial.

Reply:
But, I will import it directly from the internal/encoding rather than using the global